lumigator/backend/backend/services/jobs.py

```diff
@@ -248,34 +247,47 @@ def get_upstream_job_status(self, job_id: UUID) -> str:
         except RuntimeError as e:
             raise JobUpstreamError("ray", "error getting Ray job status", e) from e
 
-    async def on_job_complete(self, job_id: UUID, task: Callable = None, *args):
-        """Watches a submitted job and, when it terminates successfully, runs a given task.
-        Inputs:
-        - job_id: the UUID of the job to watch
-        - task: the function to be called after the job completes successfully
-        - args: the arguments to be passed to the function `task()`
+    async def wait_for_job_complete(self, job_id, max_wait_time_sec=None):
```

Comment:
Please give a sensible default for this (e.g. 5m). Otherwise we can wait arbitrarily if the job somehow goes astray.

Reply:
😆  If you're a LOTR fan you'll appreciate this meme 😂 

![image](https://github.com/user-attachments/assets/66fd7300-1845-40cb-8944-5fba85ba0a38)

I had an earlier comment from @aittalam about it being difficult to set a default wait time, due to the wide variability of how long a job may be running https://github.com/mozilla-ai/lumigator/pull/768/files/166375c48d00f642c18d41ad6d1306bb6cbac46c#r1935492921 

Similar to how the python requests lib works, I figured the default behavior would be to wait ♾️

Reply:
Yes, I remember. I don't mind a ludicrous limit - as long as there's a limit. If it can be set right now in tests, I guess that's enough for me at this point.

I assume the python requests will depend on other timeouts like TCP stack timers not to hang indefinitely. But in our case, we don't have other timers that we can rely on, afaict.

Reply:
Maybe we can indicate the last job state as retrieved from Ray, so we can see if the job has been stopped due to this timer (final state still PENDING).

Reply:
Sounds good! At the moment I'll keep this as-is since we define limits in the functions that make the calls, but this is definitely something to look into as we move forwards. I'm afraid to even suggest what I consider a ludicrous limit because then it seems like we risk setting something that we think is ludicrous but a user doesn't, and then we time out their job on them and they get frustrated 😆 . Very hypothetical of course but that's where my mind is going haha